Intersect.Server/Entities/Events/EventPageInstance.cs

```diff
@@ -446,7 +446,7 @@ protected override bool ProcessMoveRoute(Player forPlayer, long timeMs)
                                     else
                                     {
                                         //Move Randomly
-                                        moveDir = (byte)Randomization.Next(0, Options.Instance.Sprites.Directions);
+                                        moveDir = (Direction)Randomization.Next((int)Direction.None, Options.Instance.MapOpts.MovementDirections);
```

Comment:
Can you instead create an extension method for Randomization that handles directions?

Reply:
Goal: Remove multiple copies of the same casting code

Reply:
```csharp
        /// <summary>
        /// Returns a random direction from the available range of Directions.
        /// </summary>
        /// <returns>A random direction.</returns>
        public static Direction NextDirection()
        {
            lock (mRandom)
            {
                return (Direction)mRandom.Next(Options.Instance.MapOpts.MovementDirections);
            }
        }

        /// <summary>
        /// Returns a random direction within a specified range of Directions.
        /// </summary>
        /// <param name="minValue">The minimum value of the range (inclusive).</param>
        /// <param name="maxValue">The maximum value of the range (exclusive). <paramref name="maxValue" /> must be greater than or equal to <paramref name="minValue" />.</param>
        /// <returns>A Direction greater than or equal to <paramref name="minValue" /> and less than <paramref name="maxValue" />; that is, the range of return values includes <paramref name="minValue" /> but not <paramref name="maxValue" />. If <paramref name="minValue" /> equals <paramref name="maxValue" />, <paramref name="minValue" /> is returned.</returns>
        /// <exception cref="T:System.ArgumentOutOfRangeException">Thrown when <paramref name="minValue" /> is greater than <paramref name="maxValue" />.</exception>
        public static Direction NextDirection(int minValue, int maxValue)
        {
            lock (mRandom)
            {
                return (Direction)mRandom.Next(minValue, maxValue);
            }
        }
```